sdk/core/abort-controller/src/AbortController.ts

```diff
@@ -76,6 +79,7 @@ export class AbortController {
     }
     // coerce parentSignals into an array
     if (!Array.isArray(parentSignals)) {
+      // eslint-disable-next-line prefer-rest-params
       parentSignals = arguments;
```

Comment:
this code is actually wrong, since `arguments` is not an array at all. 

Isn't there something better we could do here, like use rest params and check if args[0] was an array instead?

/cc @bterlson

Reply:
Presence of `any` in the constructor signature is very unsettling also. I agree with switching to rest args here. I think a type-compatible signature for the implementation is

`constructor(...args: [any, ...AbortSignalLike[]])`

Could also use `Array.from(arguments)` to make it a true array.

Reply:
I thing using `arguments` is crucial to propagate signal information. I opened this issue to investigate further: https://github.com/Azure/azure-sdk-for-js/issues/11373.